### PR TITLE
New feature: `join`

### PR DESCRIPTION
**Pull request by noahlaux**
Introduces `join` to the built-in functions to join lists with a seperator. list items and separator can either be `string` or `number`.

Discussed in #370 & #368.

# Checklist

Before submitting a pull request, please check the following:

* [x] All tests pass for you on your machine for all implementations (all implementations must behave identically)
* [x] New tests have been added for any new functionality or to prevent regressions of a bugfix
* [x] Added a short description of the change to `newsfragments/$issue.bugfix` (for fixes) or `.feature` (for new features) or `.doc` (for documentation-only changes)

### CODE REVIEW

File: specification.yml [hunk 1]
```diff
@@ -1940,6 +1940,36 @@ context: {}
...
+---
+title: join with same
+context: {}
+template: {$eval: "join([',', ','], ',')"}
+result: ",,,"
+---
+title: join with number separator
+context: {}
+template: {$eval: "join([1, 3], 2)"}
+result: "123"
```
Comment (djmitche) [inline specification.yml:1956]: Let's add cases for the other two possibilities -- `join(["one", "two"], 3)` and `join([1, 2], "three")`.

Comment (noahlaux) [reply specification.yml:1956]: yeah good idea!

File: README.md [hunk 2]
```diff
@@ -834,6 +834,18 @@ result:
...
+#### Arrays
+
+```yaml
+template:
+  - {$eval: 'join(["carpe", "diem"], " ")'}
+  - {$eval: 'join([1, 3], 2)}'}
+context: {}
+result:
+  - carpe diem
+  - 123
```
[the remark this answers is not in the document]
Comment (noahlaux) [reply README.md:846]: Yes I'll add that to the description for arrays

**Comment by noahlaux** [comment]
Sorry I did some changes in github code editor (big mistake!)

**Pull request merged**